mosaic-site/src/components/app-header/AppHeader.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { FC, useEffect, useLayoutEffect, useState } from "react";
+import classnames from "classnames";
+import { Logo } from "@salt-ds/lab";
+import { useBreakpoint, Link } from "@jpmorganchase/mosaic-components";
+import type { TabsMenu, TabsLinkItem } from "@jpmorganchase/mosaic-components";
+import { useRoute, SidebarItem } from "@jpmorganchase/mosaic-store";
+import {
+  AppHeaderDrawer,
+  AppHeaderTabs,
+} from "@jpmorganchase/mosaic-site-components";
+import styles from "./AppHeader.module.css";
+
+export type AppHeaderProps = {
+  homeLink?: string;
+  logo?: string;
+  menu?: TabsMenu;
+  title?: string;
+};
+
+const useIsomorphicLayoutEffect =
+  typeof window !== "undefined" ? useLayoutEffect : useEffect;
+
+type MenuItem = { link: string; title?: string; links: TabsLinkItem[] };
+
+const createDrawerMenu = (menu: any[]): SidebarItem[] =>
+  menu.reduce((result: SidebarItem[], item: MenuItem) => {
+    const parsedItem = {
+      id: item.link,
+      name: item.title,
+      data: { link: item.link },
+    };
+    if (item?.links?.length) {
+      const childNodes = createDrawerMenu(item.links);
+      return [...result, { ...parsedItem, childNodes }];
+    }
+    return [...result, parsedItem];
+  }, []);
+
+export const AppHeader: FC<AppHeaderProps> = ({
+  homeLink,
+  logo,
+  menu = [],
+  title,
+}) => {
+  const [showDrawer, setShowDrawer] = useState(false);
+  const breakpoint = useBreakpoint();
+  const { route } = useRoute();
+
+  const isMobileOrTablet = breakpoint === "mobile" || breakpoint === "tablet";
+
+  useIsomorphicLayoutEffect(() => {
+    setShowDrawer(isMobileOrTablet);
+  }, [breakpoint]);
+
+  const appHeaderLogo = isMobileOrTablet ? "/img/logo_mobile.svg" : logo;
```

Comment:
In future, should aim to push this up to the props at least.

Another approach is similar to light/dark theming changes of images.
Luke is looking at a way we can load/externalize image variations, without having to add multiple URLs.
e.g light/dark, mobile, tablet, desktop, web

I did wonder whether we could move this to layouts but you still need your own styling.

I think we can assume at the moment that this AppHeader is throwaway/temporary whilst we iterate on a 'site' AppHeader that can be re-used across sites. Design folks can help here @shey-v 

What would be useful is if we can document on a wiki, what we have had to change for Salt and then we can loop back. Similar to what I did when we migrated the code to Salt and had to make some extra changes.

Reply:
Yes this is definitely a temporary component. I'm just working on the wiki now :)